Libraries/Connect/Internal/Streaming/ClientOnlyStream.swift

```diff
@@ -0,0 +1,80 @@
+// Copyright 2022-2024 The Connect Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftProtobuf
+
+/// Concrete implementation of `ClientOnlyStreamInterface`.
+///
+/// The complexity around configuring callbacks on this type is an artifact of the library
+/// supporting both callbacks and async/await. This is internal to the package, and not public.
+final class ClientOnlyStream<Input: ProtobufMessage, Output: ProtobufMessage>: @unchecked Sendable {
```

Comment:
I'm not thrilled about the necessary callback configurations in this class, but it's fairly similar to what's present in `BidirectionalAsyncStream`. The underlying reason is that since we support both classic closures/callbacks and async/await, everything is currently backed by closures/callbacks (with async/await layered on top in separate classes that are available on iOS 13+) which need to be instantiated in a specific order to work with URLSession. If we move away from callbacks at some point or invert this structure, we can simplify these